docs/fsharp-cheatsheet.md

```diff
@@ -740,6 +743,112 @@ Another way of implementing interfaces is to use *object expressions*.
+        async {
+            printfn "Created Async"
+            do! Async.Sleep (waitInSeconds * 1000)
+            printfn $"Completed Async"
+        }
+        
+    let runTask waitInSeconds =
+        task {
+            printfn "Started Task"
+            do! System.Threading.Tasks.Task.Delay (waitInSeconds * 1000)
+            printfn $"Completed Task"
+        }
+    
+    let asyncComputation = runAsync 5 // returns Async<unit> and does not print anything
+    let newTask = runTask 3  // returns System.Threading.Tasks.Task<unit> and outputs: "Started Task" <3 second delay> "Completed Task"
+
+    asyncComputation |> Async.RunSynchronously  // this now runs the async computation
+    newTask.Wait() // this will have already completed by this point
+
+    // Output:
+    // Started Task
+    // Created Async
+    // Completed Task
+    // Completed Async
+
+### Async and Task Interop
+
+As F# sits in .NET, and a lot of the codebase uses the C# async/await, the majority of actions are going to be executed and tracked using `System.Threading.Tasks.Task<'T>`s.
+
+#### Async.AwaitTask
+
+This converts a Task into an async computation. It has the [signature](#functions-signatures): `Task<'T>` -> `Async<'T>`
+
+    async {
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn $"File {fileName} has %d{bytes.Length} bytes"
+    }
+
+#### Async.StartAsTask
+
+This converts an async computation into a Task. It has the [signature](#functions-signatures): `Async<'T>` -> `Task<'T>`
+
+    async {
+        do! Async.Sleep 5000
+    } |> Async.StartAsTask
+
+### Async and the ThreadPool
+
+Below is a demonstration of the different ways to start an async computation and where it ends up in the dotnet runtime.
+In the comments; `M`, `X`, `Y`, and `Z` are used to represent differences in values.
+
+    #r "nuget:Nito.AsyncEx, 5.1.2"
+    
+    let asyncTask from =
+        async {
+            printfn $"{from} Thread Id = {System.Threading.Thread.CurrentThread.ManagedThreadId}"
+        }
+    
+    let run () =
+        printfn $"run() Thread Id = {System.Threading.Thread.CurrentThread.ManagedThreadId}"
+        asyncTask "StartImmediate" |> Async.StartImmediate      // run and wait on same thread
+        asyncTask "Start" |> Async.Start                        // queue in ThreadPool and do not wait
+        asyncTask "RunSynchronously" |> Async.RunSynchronously  // run and wait; depends
+        printfn ""
+
+    run ()
+    // run() Thread Id = M             - Main, non-ThreadPool thread
+    // StartImmediate Thread Id = M    - started, waited, and completed on run() thread
+    // Start Thread Id = X             - queued and completed in ThreadPool
+    // RunSynchronously Thread Id = Y  - started, waited, and completed in ThreadPool
+    // Important: As `Start` is queued in the ThreadPool, it might finish before `RunSynchronously` and they will share the same number
```

Comment:
there is no 'will' guarantee and its confusing to suggest there is? in general the threadpool minimizes threads used, btu thats an implementation detail, and I dont think you're serving the reader well by giving them the illusion that they have learnt all about F# Async, ThreadPools and more in one page of a cheatsheet...